### ISSUE TITLE
How to start the game without internet?

### ISSUE THREAD

**Issue by lawn9** [CmlLib/CmlLib.Core#29]
Hello there,
How do I start the game without internet? (I logged in to the crack) when I do with these codes, I get an error if I don't have internet.

[Codes.txt](https://github.com/CmlLib/CmlLib.Core/files/6811667/Codes.txt)

**and I have another problem,**
I am getting Java Virtual Machine error on 1.15+ versions. However, such an error does not occur when I use Minecraft Launcher. Could the Java version being used not support it?

(Translated)

**Comment by MelonHell** [comment]
+1
I cannot start the game when "checkAndDownload: false"

**Comment by MelonHell** [comment]
I noticed that launching is not possible until CheckAndDownloadAsync with MVersion argument is executed. This function somehow changes in MVersion, after which everything works

**Comment by MelonHell** [comment]
I think I found the problem. At least because of the JavaBinaryPath parameter in MVersion

**Comment by MelonHell** [comment]
I fixed it by using this function before running
`        private void FixJavaBinaryPath(MinecraftPath path, MVersion version)
        {
            if (!string.IsNullOrEmpty(version.JavaBinaryPath) && File.Exists(version.JavaBinaryPath))
                return;

            var javaVersion = version.JavaVersion;
            if (string.IsNullOrEmpty(javaVersion))
                javaVersion = "jre-legacy";

            version.JavaBinaryPath = Path.Combine(path.Runtime, javaVersion, "bin", MJava.GetDefaultBinaryName());
        }`

**Comment by AlphaBs** [comment]
please read this [offline-mode](https://github.com/CmlLib/CmlLib.Core/wiki/CMLauncher#offline-mode).
set FileDownloader to null, and set VersionLoader to LocalVersionLoader.

```csharp
var launcher = new CMLauncher(path);

launcher.VersionLoader = new LocalVersionLoader(launcher.MinecraftPath);
launcher.FileDownloader = null;

// ~~~

```

**Comment by MelonHell** [comment]
This does not work. The same error

**Comment by MelonHell** [comment]
At checkAndDownload: false java error, at checkAndDownload: true connection error

**Comment by AlphaBs** [comment]
fixed on [f5c8d17](https://github.com/CmlLib/CmlLib.Core/commit/f5c8d17227521f7bcb01c2ff53362556500c8f5b)

**Comment by AlphaBs** [comment]
fixed on v3.3.3
